### PR TITLE
ocpasswordgen: Hide password characters on input

### PR DESCRIPTION
**Pull request by savvamitrofanov**
This changes should hides characters on password input. In this case I use getch from conio.h library on Windows and also implemented getch for unix using unistd and termios standard libraries.

### CODE REVIEW

File: Utilities/ocpasswordgen/ocpasswordgen.c [hunk 1]
```diff
@@ -21,13 +58,34 @@ ENTRY_POINT (
...
+    if (Char == EOF || Char == CHAR_LINEFEED) {
       break;
     }
+    if (Char == CHAR_CARRIAGE_RETURN) {
+      continue;
+    }
+    if (Char == CHAR_BACKSPACE || Char == CHAR_DELETE) {
+      if (PasswordLen > 0) {
+        DEBUG ((DEBUG_ERROR, "\b \b"));
+        PasswordLen--;
```
Comment (PMheart) [inline Utilities/ocpasswordgen/ocpasswordgen.c:79]: `--PasswordLen;`

File: Utilities/ocpasswordgen/ocpasswordgen.c [hunk 2]
```diff
@@ -21,13 +58,34 @@ ENTRY_POINT (
...
+    fflush(stdin);
+    Char = getch ();
+    // Handle interrupt signal for conio
+    if (Char == CHAR_END_OF_TEXT) {
+      exit(EXIT_SUCCESS);
+    }
+    if (Char == EOF || Char == CHAR_LINEFEED) {
       break;
     }
+    if (Char == CHAR_CARRIAGE_RETURN) {
```
Comment (PMheart) [inline Utilities/ocpasswordgen/ocpasswordgen.c:73]: else if? and same below.

File: Utilities/ocpasswordgen/ocpasswordgen.c [hunk 3]
```diff
@@ -21,13 +58,34 @@ ENTRY_POINT (
   UINT8   Index;
   UINT8   PasswordHash[SHA512_DIGEST_SIZE];
 
-  DEBUG ((DEBUG_ERROR, "Please enter your password: "));
+  DEBUG ((DEBUG_ERROR, "Please enter your password: \n"));
```
Comment (vit9696) [inline Utilities/ocpasswordgen/ocpasswordgen.c:61]: In my opinion, better to switch output to printf here and leave it with printf below. DEBUG goes through a lot of intermediate buffers, and may even be disabled. Also, that extra space before `\n` is not needed.

File: Utilities/ocpasswordgen/ocpasswordgen.c (hunk 2 above)
Comment (vit9696) [reply Utilities/ocpasswordgen/ocpasswordgen.c:73]: No, since break does not return control.

File: Utilities/ocpasswordgen/ocpasswordgen.c [hunk 4]
```diff
@@ -21,13 +58,34 @@ ENTRY_POINT (
...
+    if (Char == CHAR_BACKSPACE || Char == CHAR_DELETE) {
+      if (PasswordLen > 0) {
+        DEBUG ((DEBUG_ERROR, "\b \b"));
+        PasswordLen--;
+      }
+      continue;
+    }
+
+    if (Char == ' ') {
+      DEBUG ((DEBUG_ERROR, " "));
```
Comment (vit9696) [inline Utilities/ocpasswordgen/ocpasswordgen.c:85]: I do not get it why spaces are handled differently. Password can contain spaces, and it should just display as a normal `*` in this case.

File: Utilities/ocpasswordgen/ocpasswordgen.c [hunk 5]
```diff
@@ -21,13 +58,34 @@ ENTRY_POINT (
...
+  DEBUG ((DEBUG_ERROR, "Please enter your password: \n"));
 
   for (PasswordLen = 0; PasswordLen < OC_PASSWORD_MAX_LEN; ++PasswordLen) {
-    Char = getchar ();
-    if ((Char == EOF) || (Char == '\n')) {
+    fflush(stdin);
+    Char = getch ();
+    // Handle interrupt signal for conio
+    if (Char == CHAR_END_OF_TEXT) {
+      exit(EXIT_SUCCESS);
```
Comment (vit9696) [inline Utilities/ocpasswordgen/ocpasswordgen.c:68]: Why is that success? The password was not generated, in my opinion, that's `EXIT_FAILURE`.

File: Utilities/ocpasswordgen/ocpasswordgen.c [hunk 6]
```diff
@@ -4,11 +4,48 @@
 **/
 
 #include <stdio.h>
+#ifdef _WIN32
+#include <conio.h>
```
Comment (vit9696) [inline Utilities/ocpasswordgen/ocpasswordgen.c:8]: Need to test this on Windows before merging.

File: Utilities/ocpasswordgen/ocpasswordgen.c (hunk 3 above)
Comment (PMheart) [reply Utilities/ocpasswordgen/ocpasswordgen.c:61]: Shall we use printf for all userspace tools as well?

Comment (vit9696) [reply Utilities/ocpasswordgen/ocpasswordgen.c:61]: I believe not, only here it is critical.

File: Utilities/ocpasswordgen/ocpasswordgen.c (hunk 5 above)
Comment (savvamitrofanov) [reply Utilities/ocpasswordgen/ocpasswordgen.c:68]: Yep, it should be EXIT_FAILURE

File: Utilities/ocpasswordgen/ocpasswordgen.c (hunk 6 above)
Comment (savvamitrofanov) [reply Utilities/ocpasswordgen/ocpasswordgen.c:8]: I tested it before creating pull request, it works

File: Utilities/ocpasswordgen/ocpasswordgen.c (hunk 4 above)
Comment (savvamitrofanov) [reply Utilities/ocpasswordgen/ocpasswordgen.c:85]: Definitely yes! Would be better don't print password chars at all to hide password length as we discussed before.

File: Utilities/ocpasswordgen/ocpasswordgen.c (hunk 1 above)
Comment (savvamitrofanov) [reply Utilities/ocpasswordgen/ocpasswordgen.c:79]: Thx. I'll apply uncrustify

**Pull request merged**